tests/scripts/all.sh

```diff
@@ -1,15 +1,96 @@
 #! /usr/bin/env bash
 
```

Comment:
Note that this conflicts with https://github.com/Mbed-TLS/mbedtls/pull/9293, which I hope we can merge soon since all it's missing is one approval on one backport.

Reply:
This also conflicts with https://github.com/Mbed-TLS/mbedtls/pull/9286, but that's not quite ready yet, so I suggest that we go 9293, then 9720, then 9286.

Reply:
Sorry, I'm not seeing the conflict with 9293 - it seems to me that the two PRs are not touching the same files. What am I missing?

Reply:
My bad, there's no merge conflict with 9293 and I don't think there's a semantic conflict either. There's a potential semantic conflict with https://github.com/Mbed-TLS/mbedtls/pull/9286 since it will error out if a component name is duplicated between mbedtls and tf-psa-crypto.

I ran a merge of something that conflicted but that must have been some different branches or with a non-up-to-date local copy and I can't find exactly what I ran yesterday anymore. I have a conflict checker script and it reports no conflict between the current state of this PR and any other PR opened in the last 6 months. #9293 only conflicts in `.gitignore` with a priority-medium PR.

Reply:
> There's a potential semantic conflict with #9286 since it will error out if a component name is duplicated between mbedtls and tf-psa-crypto.

Well, currently my PR also forbids component names that are duplicated between mbedtls and tf-psa-crypto, so that shouldn't be a problem. As I said earlier, I don't think we should allow components with the same name on both sides until the sides are completely separated.